### PR TITLE
Add mandatory cargo_doc

### PR DESCRIPTION
**Pull request by InfRandomness**

### CODE REVIEW

**Comment by RalfJung** [comment]
Something went wrong here, your PR contains other people's commits.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -126,15 +128,15 @@ jobs:
   end-success:
     name: bors build finished
     runs-on: ubuntu-latest
-    needs: [build, clippy]
+    needs: [build, clippy, rustdoc]
```
Comment (RalfJung) [inline .github/workflows/ci.yml:131]: This `rustdoc` job no longer exists so it needs to be removed here.

**Comment by RalfJung** [comment]
Looks good! Can you please squash this all into a single commit? Then we can land this.

**Comment by jyn514** [comment]
This doesn't seem to have caught the original CI error in rust-lang/rust.
```
warning: unresolved link to `0`
   --> src/tools/miri/src/eval.rs:400:13
    |
400 | /// in argv[0] cannot be encoded using the standard command line parsing rules.
    |             ^ no item named `0` in scope
    |
    = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
```

**Comment by jyn514** [comment]
I think you need to add `--document-private-items` to the CI job.

**Comment by InfRandomness** [comment]
> This doesn't seem to have caught the original CI error in rust-lang/rust.
> 
> ```
> warning: unresolved link to `0`
>    --> src/tools/miri/src/eval.rs:400:13
>     |
> 400 | /// in argv[0] cannot be encoded using the standard command line parsing rules.
>     |             ^ no item named `0` in scope
>     |
>     = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default
>     = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
> ```

good catch

**Comment by InfRandomness** [comment]
> I think you need to add `--document-private-items` to the CI job.

This should be ready to go now
Just would need to wait for a submodule update

**Comment by RalfJung** [comment]
This PR now has two commits with the same description? Probably better to squash them.

File: src/eval.rs [hunk 2]
```diff
@@ -403,7 +403,7 @@ pub fn eval_entry<'tcx>(
 /// The string will be UTF-16 encoded and NUL terminated.
 ///
 /// Panics if the zeroth argument contains the `"` character because doublequotes
-/// in argv[0] cannot be encoded using the standard command line parsing rules.
+/// in argv\[0\] cannot be encoded using the standard command line parsing rules.
```
Comment (jyn514) [inline src/eval.rs:406]:
```suggestion
/// in `argv[0]` cannot be encoded using the standard command line parsing rules.
```

**Comment by RalfJung** [comment]
Thanks, looking good. :)
@bors r+

**Pull request merged**